### ISSUE TITLE
twisted.python.util.mergeFunctionMetadata does not copy __module__ attribute

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3049]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#3049|
|Type|defect|
|Created|2008-02-21 17:24:21Z|
|Branch|https://github.com/twisted/twisted/tree/mergemodule-3049|

This is an oversight.  `twisted.python.deprecate` currently manually accounts for this, but it is doubtful that other users do.

<details><summary>Searchable metadata</summary>

```
trac-id__3049 3049
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_mergemodule_3049 branches/mergemodule-3049
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1203614661000000 1203614661000000
changetime__1204237155000000 1204237155000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_1"></a>|@jml commented|
|-|-|

Oh, duh. I made some fixes `mergeFunctionMetadata` in the `deprecate` branch, but for some reason didn't twig that `__module__` should also be copied.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [22714]) Branching to 'mergemodule-3049'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve commented|
|-|-|

Looks good, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [22720]) Merge mergemodule-3049

Author: exarkun
Reviewer: therve
Fixes [#3049](https://github.com/twisted/twisted/issues/3049)

Copy the __module__ attribute from the one function to the other in
mergeFunctionMetadata.
